### DISCUSSION TITLE
Browsershot no longer works with Puppeteer v23

### DISCUSSION THREAD

**Discussion by PHLAK** [spatie/browsershot#875] [Bugs]
Puppeteer recently released [v23 with some breaking changes](https://github.com/puppeteer/puppeteer/releases/tag/puppeteer-v23.0.0) and Browsershot no longer works with this version. I even tried Browsershot v4.2.0 witch includes #870 that claims to handle output for Puppeteer v23 and it also didn't work (though differently).

I suspect the main culprit of this is https://github.com/puppeteer/puppeteer/pull/12757 where all references to `browser` were replaced with `product` in Puppeteer. This likely requires some changes to Browsershot to accommodate the changes.

**Comment by bluesheep100** [comment]
I'm trying to replicate the issue of generated PDFs being blank, but the test cases all produce PDFs with content just fine. (I just removed the cleanup to see them)
I've attempted this on Puppeteer 22.15.0 (latest before 23) and 23.0.2, but browsershot 4.2.0 works fine with both. This is running inside an Ubuntu 22.04 LTS distro on WSL2 for Window 11, and works regardless of whether my Puppeteer install is local or global.

It's strange that I can't replicate this bug, as I experienced it myself while testing #870, but I can't recall how I got around it.
It could be related to the packages listed as dependencies if you're running on a Forge provisioned server, as seen [here](https://spatie.be/docs/browsershot/v4/requirements)

The install command I ran for my own environment, and in Sail's Dockerfile:
```
sudo apt-get install -y gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget libgbm-dev libxshmfence-dev
```
Please give everything you can about your specific environment, including versions of Puppeteer, Chrome, Nodejs, running on Browsershot 4.2.0, and include some test code.
